pkg/controller/node/status.go

```diff
@@ -34,14 +34,12 @@ func (ctrl *Controller) syncStatusOnly(pool *mcfgv1.MachineConfigPool) error {
 }
 
 func calculateStatus(pool *mcfgv1.MachineConfigPool, nodes []*corev1.Node) mcfgv1.MachineConfigPoolStatus {
-	previousUpdated := mcfgv1.IsMachineConfigPoolConditionTrue(pool.Status.Conditions, mcfgv1.MachineConfigPoolUpdated)
-
 	machineCount := int32(len(nodes))
 
-	updatedMachines := getUpdatedMachines(pool.Status.Configuration.Name, nodes)
+	updatedMachines := getUpdatedMachines(pool.Spec.Configuration.Name, nodes)
```

Comment:
if the `Spec` of an object describes the desired state, shouldn't this routine use Status? we're interested in nodes at a given state (not the desired) - I may be missing something and re-reading https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status

Reply:
Hmm.  We're trying to calculate the `Updated`/`Updating` conditions here, so we need to know how many nodes are at our *desired* state.

I'll admit to not being steeped in the art here - can you think of anything else similar to this?

But, it would feel very strange if these values followed Status - the `updatedMachineCount` would go *down* as we updated...

Reply:
> I'll admit to not being steeped in the art here - can you think of anything else similar to this?

the template controller is the only one I can think of - it uses Status to tell the render ctrl "I'm done"

but I believe you're right and we're just calculating where we are wrt desired state here.

Reply:
Another way to think of this is; we probably should never have had `Configuration.Name` as part of `Status`, it should have always been in `Spec`. 

Having parts of `Status` should be derived from other `Status` fields seems wrong.

In effect, the `Status.Configuration.Name` is more like what the CVO is doing with its `history` field.

Yet another argument; `Status.Configuration` is almost meaningless given that multiple configurations could have been rolled out.  If any API client wants to know the state of the system, there's really either "fully updated" or "updating".  Any more detail on "updating" really requires looking at each node for its `currentConfig`.  The `Status.Configuration` isn't so useful for that.